opendse-io/src/main/java/net/sf/opendse/io/ReaderTGFF.java

```diff
@@ -0,0 +1,575 @@
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import net.sf.opendse.model.Application;
+import net.sf.opendse.model.Communication;
+import net.sf.opendse.model.Dependency;
+import net.sf.opendse.model.Link;
+import net.sf.opendse.model.LinkTypes;
+import net.sf.opendse.model.Mapping;
+import net.sf.opendse.model.Mappings;
+import net.sf.opendse.model.Resource;
+import net.sf.opendse.model.ResourceTypes;
+import net.sf.opendse.model.SpecificationTypeBased;
+import net.sf.opendse.model.Task;
+
+/**
+ * The {@code ReaderTGFF} imports an {@code Application}, {@code Mappings} and a
+ * set of {@code Resource} types from a .tgff-file, as generated by Task Graphs
+ * For Free (TGFF).
+ * 
+ * In particular, it serves to import benchmark task graphs, processor types and
+ * corresponding mapping possibilities from the Embedded Systems Synthesis
+ * Benchmarks Suite (E3S). See <a href=
+ * "http://ziyang.eecs.umich.edu/~dickrp/e3s/">http://ziyang.eecs.umich.edu/~dickrp/e3s/</a>
+ * 
+ * @author Valentina Richthammer
+ */
+public class ReaderTGFF {
+
+	public static final String TGFF_TYPE = "TGFF_TYPE";
+	public static final String PERIOD = "PERIOD";
+	public static final String MSG_SIZE = "MSG_SIZE";
+	public static final String RES_ATTRIBUTES = "RES_ATTRIBUTES";
+	public static final String RES_VALUES = "RES_VALUES";
+	public static final String HARD_DEADLINE = "HARD_DEADLINE";
+	public static final String SOFT_DEADLINE = "SOFT_DEADLINE";
+	public static final String WIRE = "@WIRING";
+
+	protected static final String HYPERPERIOD = "@HYPERPERIOD";
+	protected static final String TASK_GRAPH = "@TASK_GRAPH";
+	protected static final String COMMUN_QUANT = "@COMMUN_QUANT";
+	protected static final String CORE = "@CORE";
+	protected static final String CLIENT_PE = "@CLIENT_PE";
+	protected static final String SERVER_PE = "@SERVER_PE";
+	protected static final String PROC = "@PROC";
+
+	protected static final String LINK = "LINK";
+	protected static final String TASK = "TASK";
+	protected static final String ARC = "ARC";
+
+	protected static final String VALID = "valid";
+	protected static final String TYPE = "type";
+
+	protected static final String CLOSING = "}";
+	protected static final String COMMENT = "#";
+	protected static final String AT = "@";
+	protected static final String SEPARATOR = "\\s+";
+	protected static final String CONNECTOR = "_";
+	protected static final String HEADER = "#---------";
+
+	protected Map<String, String> properties;
+	protected Map<String, Double> messageSizes;
+	protected double hyperperiod;
+
+	protected Map<String, List<Task>> tgffTypeMap = new HashMap<String, List<Task>>();
+
+	/**
+	 * Reads a specification from a file.
+	 * 
+	 * @param filename
+	 *            the file name
+	 * @return the specification
+	 */
+	public SpecificationTypeBased read(String filename) {
+		return read(new File(filename));
+	}
+
+	/**
+	 * Reads a specification from a file.

```

Comment:
`Reads a {@link SpecificationTypeBased} from a {@link File}.`

and so on. Then, comments will be also adapted, if we choose to rename a class. Moreover, in Javadoc hyperlinks are created and your IDE (e.g. Eclipse) shows information about the linked class.